app.js

```diff
@@ -17,20 +18,26 @@ app.get("/api/articles/:article_id", getArticleById);
 
 app.get("/api/articles/:article_id/comments", getCommentsByArticleId);
 
-app.use((err, req, res, next) => {
-  if (err.code === "22P02") {
-    res.status(400);
-    res.send({ msg: "bad request" });
-  } else {
-    next(err);
-  }
-});
+app.post("/api/articles/:article_id/comments", postCommentByArticleId);
 
 app.use((err, req, res, next) => {
   if (err.msg) {
     res.status(err.status);
     res.send({ msg: err.msg });
   }
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.code === "22P02" || err.code === "23502") {
+    console.log(err);
+    res.status(400);
+    res.send({ msg: "bad request" });
+  } else if ((err.code = "23503")) {
```

Comment:
This Error handler is totes for psql errors but you could split this functionality out further into 2 functions 1 to handle psql 400s and one to handle psql404s